core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/pagination/PaginationSP.java

```diff
@@ -114,29 +128,56 @@ public String printIndex(String javascriptFunc) {
       if (!isLastPage()) {
         // display next page link
         int index = getIndexForNextPage();
-        result.append("<div class=\"pageOff\">");
+        result.append("<a class=\"pageOff\"");
         result.append(getLink(javascriptFunc, index, getString("GEF.pagination.nextPage")));
         result.append("<img src=\"").append(getIconsPath()).append(
-            "/arrows/arrowRight.gif\" border=\"0\" align=\"absmiddle\" alt=\"")
-            .append(getString("GEF.pagination.nextPage")).append("\"/></a>");
-        result.append("</div>");
+            "/arrows/arrowRight.png\" border=\"0\" align=\"absmiddle\" alt=\"")
+            .append(getString("GEF.pagination.nextPage")).append("\"/>");
+        result.append("</a>");
 
         if (currentPage + getNumberOfPagesAround() < nbPage) {
           // display last page link
-          result.append("<div class=\"pageOff\">");
+          result.append("<a class=\"pageOff\"");
           result.append(getLink(javascriptFunc, getIndexForLastPage(),
               getString("GEF.pagination.lastPage")));
           result.append("<img src=\"").append(getIconsPath()).append(
-              "/arrows/arrowDoubleRight.gif\" border=\"0\" align=\"absmiddle\" alt=\"")
-              .append(getString("GEF.pagination.lastPage")).append("\"/></a> ");
-          result.append("</div>");
+              "/arrows/arrowDoubleRight.png\" border=\"0\" align=\"absmiddle\" alt=\"")
+              .append(getString("GEF.pagination.lastPage")).append("\"/>");
+          result.append("</a>");
         }
       }
+      result.append("</div>");
 
+      if (nbItemsPerPage && getNbItems() > 25) {
+        result.append("<div class=\"pageIndex numberPerPage\">");
+        int[] values = new int[]{25, 50, 100};
+        for (int i=0; i<values.length; i++) {
+          int value = values[i];
+          if (getNbItems() > value || (i != 0 && getNbItems() > values[i-1])) {
+            if (getNbItemsPerPage() == value) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS134)